.github/workflows/paths-ignore.yaml

```
@@ -0,0 +1,80 @@
+# This is a reusable workflow to determine if the current change requires an E2E
+# run.  This is required because using [paths-ignored] directly means the whole
+# workflow is skipped, but that means that it doesn't count as having run a
+# required workflow.
+
+# Usage:
+#   jobs:
+#     check_paths:
+#        uses: ./.github/workflows/actions/paths-ignore.yaml
+#     do_thing:
+#        if: jobs.check_paths.outputs.should_run == 'true'
+#        # Unfortunately, a string comparison is required.
+
+name: Check for ignored paths
+
+on:
+  workflow_call:
+    inputs:
+      paths_ignore:
+        description: Paths to ignore.
+        type: string
+        default: |
+          - '.github/actions/spelling/**'
+          - 'bats/**'
+          - 'docs/**'
+          - '**.md'
```

Why does it use this stringified YAML syntax. Couldn't it be simply a list of strings (drop the leading `- ` prefix)? Wouldn't that also make it easier to parse? Also not sure if we need the quotes if the rule is one pattern per line:

```suggestion
        default: |
          .github/actions/spelling/**
          bats/**
          docs/**
          **.md
```

Now that I realize that it requires a subtly different input format, I wonder if it wouldn't be better to change the input name to something like `path-ignore-list`?